DataManagementSystem/DB/FileCatalogDB.py

```diff
@@ -70,22 +58,69 @@ def setConfig(self, databaseConfig):
     self.visibleReplicaStatus = databaseConfig['VisibleReplicaStatus']
 
-    try:
-      # Obtain the plugins to be used for DB interaction
-      self.ugManager = eval("%s(self)" % databaseConfig['UserGroupManager'])
-      self.seManager = eval("%s(self)" % databaseConfig['SEManager'])
-      self.securityManager = eval("%s(self)" % databaseConfig['SecurityManager'])
-      self.dtree = eval("%s(self)" % databaseConfig['DirectoryManager'])
-      self.fileManager = eval("%s(self)" % databaseConfig['FileManager'])
-      self.datasetManager = eval("%s(self)" % databaseConfig['DatasetManager'])
-      self.dmeta = eval("%s(self)" % databaseConfig['DirectoryMetadata'])
-      self.fmeta = eval("%s(self)" % databaseConfig['FileMetadata'])
-    except Exception as x:
-      gLogger.fatal("Failed to create database objects", x)
-      return S_ERROR("Failed to create database objects")
+    # Obtain the plugins to be used for DB interaction
+    self.objectLoader = ObjectLoader()
+
+    result = self.__loadCatalogComponent(databaseConfig['UserGroupManager'])
+    if not result['OK']:
+      return result
+    self.ugManager = result['Value']
+
+    result = self.__loadCatalogComponent(databaseConfig['SEManager'])
+    if not result['OK']:
+      return result
+    self.seManager = result['Value']
+
+    result = self.__loadCatalogComponent(databaseConfig['SecurityManager'])
+    if not result['OK']:
+      return result
+    self.securityManager = result['Value']
+
+    result = self.__loadCatalogComponent(databaseConfig['DirectoryManager'])
+    if not result['OK']:
+      return result
+    self.dtree = result['Value']
+
+    result = self.__loadCatalogComponent(databaseConfig['FileManager'])
+    if not result['OK']:
+      return result
+    self.fileManager = result['Value']
+
+    result = self.__loadCatalogComponent(databaseConfig['DatasetManager'])
+    if not result['OK']:
+      return result
+    self.datasetManager = result['Value']
+
+    result = self.__loadCatalogComponent(databaseConfig['DirectoryMetadata'])
+    if not result['OK']:
+      return result
+    self.dmeta = result['Value']
+
+    result = self.__loadCatalogComponent(databaseConfig['FileMetadata'])
+    if not result['OK']:
+      return result
+    self.fmeta = result['Value']
 
     return S_OK()
 
+  def __loadCatalogComponent(self, componentName):
+    """ Create an object of a given catalog component
+    """
+    moduleName = componentName
+    # some modules contain several implementation classes
+    for m in ['SEManager', 'UserAndGroupManager', 'SecurityManager']:
+      if m in componentName:
+        moduleName = m
+    componentPath = 'DataManagementSystem.DB.FileCatalogComponents'
+    result = self.objectLoader.loadObject('%s.%s' % (componentPath, moduleName),
+                                          componentName.split('.')[-1])
+    if not result['OK']:
+      gLogger.error('Failed to load catalog component', result['Message'])
```

Comment:
I think it would be nice that in case of error you print also the option key that lead to the problem, and not just the message from the object loader, which might be more cryptic.
That would mean passing the databaseConfig and the key used to access the value separately I guess.

Reply:
Updated